src/firstSuggestionDecoration.ts

```diff
@@ -0,0 +1,120 @@
+  setFirstSuggestionFlag,
+} from "./firstSuggestionDecorationFlag";
+import { fireEvent, getState } from "./binary/requests/requests";
+import isInTheLastHour from "./utils/time.utils";
+import { getTabnineExtensionContext } from "./globals/tabnineExtensionContext";
+import { setDecoration } from "./vscode.api";
+import TabnineInlineCompletionItem from "./inlineSuggestions/tabnineInlineCompletionItem";
+import { MAX_SMALL_INTEGER_V8 } from "./globals/consts";
+import { Capability, isCapabilityEnabled } from "./capabilities/capabilities";
+// Anchor hint to the right of the line
+export const firstSuggestionDecorationType = window.createTextEditorDecorationType(
+  {
+    rangeBehavior: DecorationRangeBehavior.ClosedOpen,
+  }
+);
+
+export function getFirstCompletionDecoration(
+  document: TextDocument,
+  position: Position
+): DecorationOptions[] {
+  const decorations: DecorationOptions[] = [];
+  decorations.push({
+    renderOptions: {
+      after: {
+        color: "gray",
+        contentText: "👈 Press Tab to accept Tabnine's suggestion",
+        margin: "20px",
+        border: "0.5px solid",
+      },
+    },
+    range: document.validateRange(
+      new Range(
+        position.line,
+        MAX_SMALL_INTEGER_V8,
+        position.line,
+        MAX_SMALL_INTEGER_V8
+      )
+    ),
+  });
+  return decorations;
+}
+
+export function displayFirstSuggestionDecoration(position: Position): void {
+  const editor = window.activeTextEditor;
+  if (!editor) {
+    return;
+  }
+  setDecoration(
+    firstSuggestionDecorationType,
+    getFirstCompletionDecoration(editor.document, position),
+    editor
+  );
+  // for now: clear notification after 5 seconds
+  setTimeout(clearFirstSuggestionDecoration, 5000, editor);
+}
+
+// message should be displayed if it has not been displayed yet,
+// and if the extension has been installed within the last hour
+export async function shouldDisplayFirstSuggestionDecoration(
+  context: ExtensionContext
+): Promise<boolean> {
+  if (
+    !isCapabilityEnabled(Capability.FIRST_SUGGESTION_DECORATION) ||
+    firstSuggestionDecorationAlreadyDisplayed(context)
+  ) {
+    return false;
+  }
+  const state = await getState();
+
+  return !!(
+    state?.installation_time &&
+    isInTheLastHour(new Date(state?.installation_time))
+  );
+}
+
+function isEmptyCompletion(suggestion: TabnineInlineCompletionItem): boolean {
+  if (!suggestion.insertText) {
+    return true;
+  }
+  return suggestion.insertText.trim() === "";
+}
+
+export async function handleFirstSuggestionDecoration(
+  position: Position,
+  completions: InlineCompletionList
+): Promise<void> {
+  if (
+    completions.items.length === 0 ||
+    isEmptyCompletion(completions.items[0])
+  ) {
+    return;
+  }
+  const context = getTabnineExtensionContext();
+  if (context && (await shouldDisplayFirstSuggestionDecoration(context))) {
+    displayFirstSuggestionDecoration(position);
+    await Promise.all([
```

Comment:
i think you definitely want to `await setFirstSuggestionFlag` but probably `void fireEvent`...